apps/client/src/app/auth/_state/server/hooks/usePostSignUp.ts

```diff
@@ -0,0 +1,28 @@
+import { useModal } from "@shophub/ui";
+import { useMutation } from "@tanstack/react-query";
+import type { SignUpFormValues } from "~/app/auth/_types";
+import { API_SIGN_UP } from "../api";
+
+export const usePostUserSignUp = () => {
+  const modal = useModal();
+
+  return useMutation(
+    async (userInfo: SignUpFormValues) => {
+      await API_SIGN_UP.Request(userInfo);
+    },
+    {
+      onSuccess: () =>
+        modal.open({
+          title: "회원가입이 완료되었습니다.",
+          type: "positive",
+          onConfirm: () => alert("회원가입 완료"),
+        }),
+      onError: (error) =>
+        modal.open({
+          title: "회원가입을 실패하였습니다.",
+          type: "negative",
+          onConfirm: () => alert(error),
+        }),
+    },
+  );
```

Comment:
저는 객체로 작성하는 거 선호하긴 하는데 어떠신가요 ? `mutationFn` 으로 명시적으로 넣어줬어요!
아, 그리고 `mutationFn: (userInfo: SignUpFormValues) => API_SIGN_UP.Request(userInfo)`처럼 async/await 떼도 정상 동작합니다!

테스트할 겸 요청 보내봤는데, 회원가입 성공하고 어디론가 보내주면(ex. 로그인하러?) 좋을 거 같아요~~!

그리고!! Modal 조금 수정한 거 반영했어용 onConfirm 보내지 않으면 확인 눌렀을 때 닫히는 액션만 하게 됩니당

```suggestion
  return useMutation({
    mutationFn: (userInfo: SignUpFormValues) => API_SIGN_UP.Request(userInfo),
    onSuccess: () =>
      modal.open({
        title: "회원가입이 완료되었습니다.",
        type: "positive",
        onConfirm: () => alert("회원가입 완료"),
      }),
    onError: (error) =>
      modal.open({
        title: "회원가입을 실패하였습니다.",
        type: "negative",
        onConfirm: () => alert(error),
      }),
  });
```

Reply:
이거는 제가 다른 리뷰에도 남겼는데, react-query 버전업이 되면 사용할 수 없는 인터페이스라 미리 객체로 다 변환하는 것도 좋을 것 같아요, eslint 도 추가하는 것을 추천드립니다
https://github.com/shophubkr/shophub-client/pull/52#discussion_r1473720801